File: BackEnd/estagio/src/main/java/com/utfpr/estagio/dto/EstudanteDto.java
package com.utfpr.estagio.dto;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class EstudanteDto {
    
    private String nomeEstudante;
    private List<PeriodoEstagioDto> periodosEstagio;

    @Data
    @Builder
    public static class PeriodoEstagioDto {
        @JsonFormat(pattern = "dd/MM/yyyy")
        private LocalDate dataInicioEstagio;
        
        @JsonFormat(pattern = "dd/MM/yyyy")
        private LocalDate dataTerminoEstagio;
        
        private List<DatasRelatoriosParciaisAlunoDto> datasRelatoriosParciaisAluno;
        
        private List<DatasRelatoriosParciaisOrientadorDto> datasRelatoriosParciaisOrientador;
        
        private AvaliacaoDto intervaloAvaliacao;
        
        @JsonFormat(pattern = "dd/MM/yyyy")
        private LocalDate dataRelatorioVisita;
        
        @Builder.Default
        private boolean enviadoRelatorioVisita = false;

        @JsonFormat(pattern = "dd/MM/yyyy")
        private LocalDate dataRelatorioFinal; 
        
        @Builder.Default
        private boolean enviadoRelatorioFinal = false;
        
        private Boolean obrigatorio;
        
        private String empresa;
        
        
    }
}

File: FrontEnd/estagio-utfpr/src/app/models/estudante.dto.ts
export interface PeriodoRelatorio {
  inicioPeriodo: Date;
  fimPeriodo: Date;
}

export interface EstudanteDto {
  nomeEstudante: string;
  periodosEstagio: PeriodoEstagioDto[];
}
export interface PeriodoEstagioDto {
    dataInicioEstagio: Date;
    dataTerminoEstagio: Date;
    datasRelatoriosParciaisAluno: DatasRelatoriosParciaisAlunoDto[];
    datasRelatoriosParciaisOrientador: DatasRelatoriosParciaisOrientadorDto[];
    dataRelatorioVisita?: Date;
    enviadoRelatorioVisita?: boolean;
    dataRelatorioFinal: Date;
    enviadoRelatorioFinal?: boolean;
    intervaloAvaliacao?: AvaliacaoDto;
    empresa?: String;
    obrigatorio: boolean;
}

export interface DatasRelatoriosParciaisAlunoDto {
    inicioPeriodoRelatorio: Date;
    fimPeriodoRelatorio: Date;
    enviado: boolean;
}

export interface DatasRelatoriosParciaisOrientadorDto {
    inicioPeriodoRelatorio: Date;
    fimPeriodoRelatorio: Date;
    enviado: boolean;
}

export interface AvaliacaoDto {
    dataInicio: Date;
    dataFim: Date;
}

export interface RelatorioVisitaDto {
    dataLimite: Date;
    enviado: boolean;
}

export interface RelatorioFinalDto {
    dataLimite: Date;
    enviado: boolean;
}
